### DISCUSSION TITLE
Realtime (self-hosted) config: where to set disconnectOnNoSubscriptions = false?

### DISCUSSION THREAD

**Discussion by sinameraji** [supabase/supabase#30187] [Questions]
Hello,

I'm using supabase realtime (i'm self hosting supabase) in my next js app. i have a use case where i wanna maintain 1 websocket connection that joins and leaves different channels, making sure 1 channel is active at a time, without disconnecting the websocket connection as it does these phx joins and leaves.

the problem i'm facing is that when i leave a channel, it currently disconnects websocket, and creates a new connection and then joins the new channel. i noticed a hint in supabase realtime docs for Swift that this might be caused by disconnectOnNoSubscriptions being set to true in realtime config. It seems to me that this creates a higher CPU usage than if i had 1 connection per supabase client, which then joined and left different channels.

how/where can i set disconnectOnNoSubscriptions to false? do i add it to my supabase docker compose yml file? if yes, will i just add it as an env variables? with the same spelling?

if i'm on the wrong track, i appreciate a heads up too. thank you
![CleanShot 2024-10-30 at 15 03 18@2x](https://github.com/user-attachments/assets/0e31d9d9-c265-483c-8789-5c4e644fb2ea)

**Comment by GaryAustin1** [comment]
Try adding a similar realtime section with that flag here:
![image](https://github.com/user-attachments/assets/caa8f038-382a-447b-bd04-1144f4c1688f)

**Reply by sinameraji** [reply]
where exactly would i put the flag? and would i keep all these other options?

**Reply by GaryAustin1** [reply]
I don't know the Swift code so just guessing.
No, you would not need other options.
I suggest:
```
realtime: .init(
  disconnectOnNoSubscriptions:false
  )
  ```
